src/validator.rs

```diff
@@ -10,3 +14,9 @@ pub trait ValidatorCode<D, R> {
     fn execute(&self, datum: D, redeemer: R, ctx: TxContext) -> Result<()>;
```

Comment:
This could return `Return<(), ValidatorCodeError>` instead. Kinda like above convo.